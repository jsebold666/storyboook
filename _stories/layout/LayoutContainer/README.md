The `<LayoutContainer>` component is used for centering containers and for responsive grid columns. The grid system works exactly like Twitter Bootstrap's, with columns automatically stacking at mobile widths. All page content except for the gds-page-header and gds-slide-nav components should be placed inside the `<LayoutContainer />`.

By default, the LayoutContainer has a max-width of 1200px. For a 100% width container, use the attribute `full={ true }`,
