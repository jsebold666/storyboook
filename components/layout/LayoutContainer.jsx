import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const LayoutContainer = ({ className, fullWidth, children, ...props }) => {
  const baseClass = 'gds-layout__container';
  const rootClass = cx(baseClass, className, {
    [`${baseClass}--full-width`]: fullWidth
  });

  return (
    <div className={rootClass} {...props}>
      {children}
    </div>
  );
};

LayoutContainer.displayName = 'LayoutContainer';

LayoutContainer.defaultProps = {
  fullWidth: false
};

LayoutContainer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  fullWidth: PropTypes.bool
};

export default LayoutContainer;
