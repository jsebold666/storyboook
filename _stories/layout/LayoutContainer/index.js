import React from 'react';
import { boolean, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import Column from '../../../components/layout/Column';
import LayoutContainer from '../../../components/layout/LayoutContainer';
import Row from '../../../components/layout/Row';

const component = () => (
  <LayoutContainer
    fullWidth={boolean('Full width', false)}
    id={text('Ids')}
    onClick={action('layout_container_clicked')}
    className={text('Extra classes')}>
    <Row>
      <Column md={4}>Example</Column>

      <Column md={4}>Example</Column>
      <Column md={4}>Example</Column>
    </Row>
  </LayoutContainer>
);

export default [readme, component];
