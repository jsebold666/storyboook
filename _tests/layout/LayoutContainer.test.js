/* globals mount */
import React from 'react';
import LayoutContainer from '../../components/layout/LayoutContainer';

describe('Expect <LayoutContainer>', () => {
  it('to render', () => {
    const props = {
      className: 'cool-stuff'
    };
    const wrapper = mount(
      <LayoutContainer {...props}>
        <div>Content</div>
      </LayoutContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('to render as full width', () => {
    const props = {
      className: 'cool-stuff',
      fullWidth: true
    };
    const wrapper = mount(
      <LayoutContainer {...props}>
        <div>Content</div>
      </LayoutContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
