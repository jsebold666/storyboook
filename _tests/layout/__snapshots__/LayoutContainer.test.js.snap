// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LayoutContainer> to render 1`] = `
<LayoutContainer
  className="cool-stuff"
  fullWidth={false}
>
  <div
    className="gds-layout__container cool-stuff"
  >
    <div>
      Content
    </div>
  </div>
</LayoutContainer>
`;

exports[`Expect <LayoutContainer> to render as full width 1`] = `
<LayoutContainer
  className="cool-stuff"
  fullWidth={true}
>
  <div
    className="gds-layout__container cool-stuff gds-layout__container--full-width"
  >
    <div>
      Content
    </div>
  </div>
</LayoutContainer>
`;
